boot/systems.go

```diff
@@ -447,3 +448,38 @@ func DropRecoverySystem(dev Device, systemLabel string) error {
 	const expectReseal = true
 	return resealKeyToModeenv(dirs.GlobalRootDir, m, expectReseal)
 }
+
+// AddRecoveryCapableSystem records a given system as one that we can recover
+// from.
+func AddRecoveryCapableSystem(systemLabel string) error {
+	opts := &bootloader.Options{
+		// setup the recovery bootloader
+		Role: bootloader.RoleRecovery,
+	}
+	// TODO:UC20: seed may need to be switched to RW
+	bl, err := bootloader.Find(InitramfsUbuntuSeedDir, opts)
+	if err != nil {
+		return err
+	}
+	rbl, ok := bl.(bootloader.RecoveryAwareBootloader)
+	if !ok {
+		return nil
+	}
+	vars, err := rbl.GetBootVars("snapd_recovery_capable_systems")
+	if err != nil {
+		return err
+	}
+	var systems []string
+	if vars["snapd_recovery_capable_systems"] != "" {
+		systems = strings.Split(vars["snapd_recovery_capable_systems"], ",")
+		if strutil.ListContains(systems, systemLabel) {
+			// already recorded, nothing to do
+			return nil
+		}
+	}
+	systems = append(systems, systemLabel)
```

Comment:
should we prepend here like we do for seeded-systems in state?

Reply:
@bboozzoo did you see this question?

Reply:
I left appending like we do with good recovery systems in modeenv. I think that in either case the bootloader handling will unfortunately need to be smart about handling this properly. I can tweak the code to prepend the system.

Reply:
no, it's fine to be consistent with modeenv I think, just wanted to check